decidim-core/db/migrate/20240722215500_change_object_changes_on_versions.rb

```diff
@@ -5,32 +5,14 @@ def up
     rename_column :versions, :object_changes, :old_object_changes
     add_column :versions, :object_changes, :jsonb # or :json
 
-    PaperTrail::Version.where.not(old_object_changes: nil).find_each do |version|
-      # This is an adaptation of PaperTrail internal load_changeset method,having in mind that we
-      # need to call also the code from PaperTrail::AttributeSerializer::ObjectChangesAttribute
-      object_changes = ActiveSupport::HashWithIndifferentAccess.new(YAML.unsafe_load(version.old_object_changes))
-      unless version.item_type.constantize.unscoped.find_by(id: version.item_id).nil?
-        # This is the deserialization code from `PaperTrail::AttributeSerializer::ObjectChangesAttribute`
-        # where we skip checking the object changeset column type, as we migrate it from YAML to JSON
-        changes_to_serialize = object_changes.clone
-        if changes_to_serialize.present?
-          serializer = PaperTrail::AttributeSerializers::CastAttributeSerializer.new(version.item_type.constantize)
-          changes_to_serialize.each do |key, change|
-            # `change` is an Array with two elements, representing before and after.
-            object_changes[key] = Array(change).map do |value|
-              serializer.send(:deserialize, key, value)
-            end
-          end
-        end
-      end
+    PaperTrail::Version.reset_column_information
 
-      version.update_columns(old_object_changes: nil, object_changes:) # rubocop:disable Rails/SkipsModelValidations
+    PaperTrail::Version.where.not(old_object_changes: nil).find_each do |version|
+      # we set a time interval to ensure that deployment has succeeded
+      Decidim::Migrate::PaperTrailJob.set(wait: 10.minutes).perform_later(version.id)
     rescue NameError
```

Comment:
This `rescue` is not relevant here. We should move it to the job where it's being raised now.

Reply:
This is meant to prevent issues with the historic models that we've removed ( Decidim::Consultation, Decidim::Election etc )

Reply:
done in [27ba262](https://github.com/decidim/decidim/pull/14139/commits/27ba26292f5a301801b32c05234e0e8d8ef1a8bd)